core/src/test/java/org/elasticsearch/index/query/FilteredQueryBuilderTest.java

```diff
@@ -0,0 +1,85 @@
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.ConstantScoreQuery;
+import org.apache.lucene.search.Query;
+import org.elasticsearch.common.lucene.search.Queries;
+
+import java.io.IOException;
+
+@SuppressWarnings("deprecation")
+public class FilteredQueryBuilderTest extends BaseQueryTestCase<FilteredQueryBuilder> {
+
+    @SuppressWarnings("rawtypes")
+    @Override
+    protected FilteredQueryBuilder createTestQueryBuilder() {
+        QueryBuilder queryBuilder = null;
+        if (frequently()) {
+            queryBuilder = RandomQueryBuilder.create(random());
+        }
+        QueryBuilder filterBuilder = null;
+        if (frequently()) {
+            filterBuilder = RandomQueryBuilder.create(random());
+        }
+
+        FilteredQueryBuilder query = new FilteredQueryBuilder(queryBuilder, filterBuilder);
+        if (randomBoolean()) {
+            query.boost(2.0f / randomIntBetween(1, 20));
+        }
+        if (randomBoolean()) {
+            query.queryName(randomUnicodeOfLengthBetween(3, 15));
+        }
+        return query;
+    }
+
+    @Override
+    protected Query createExpectedQuery(FilteredQueryBuilder qb, QueryParseContext context) throws IOException {
+        // parsed internally, but returned null during parsing...
+        if (qb.query() == null) {
+            return null;
+        }
+
+        if (qb.filter() == null) {
+            return qb.query().toQuery(context);
+        }
+
+        Query query = qb.query().toQuery(context);
+        Query filter = qb.filter().toQuery(context);
+
+        if (Queries.isConstantMatchAllQuery(filter)) {
+            // this is an instance of match all filter, just execute the query
+            return qb.query().toQuery(context);
+        }
+
+        // if its a match_all query, use constant_score
+        if (Queries.isConstantMatchAllQuery(query)) {
+            Query q = new ConstantScoreQuery(filter);
+            q.setBoost(qb.boost());
+            return q;
+        }
+
+        BooleanQuery filteredQuery = Queries.filtered(qb.query().toQuery(context), filter);
+        filteredQuery.setBoost(qb.boost());
+        return filteredQuery;
```

Comment:
don't worry too much about this method, we wil get rid of it soon. I am preparing the field for this change :)